ios/events/RNSHeaderHeightChangeEvent.mm

```diff
@@ -31,6 +31,11 @@ - (NSDictionary *)body
   return body;
 }
 
+- (BOOL)canCoalesce
+{
+  return NO;
```

Comment:
Actually don't we want this event to be coalesced? If we spam-emit the events from native to JS (e.g. when using scrollview & stretching large header) wouldn't it help improving performance a bit? 
I'm not familiar with event emitting internals & I'm not sure what are the conditions for the event instances to be coalesced, but it might be worth checking.

Reply:
You're right, I think we want this event to coalesce, but I'm afraid if there will be some issues with parsing double to `uint16_t` - I just wanted to make this change as safe as possible, but I can add coalescing key to this event 👍

Reply:
Why do you need uint16? Coalescing key?

Reply:
Resolved in https://github.com/software-mansion/react-native-screens/pull/1933/commits/db7cd697d2336038b22c5d0bffafd5990369a7bc. Thanks!

One note from me about `coalesceWIthEvent` method - I believe both `coalescingKey` and `coalesceWithEvent` methods are required here, even if we're just returning `newEvent` from argument.

<img width="511" alt="image" src="https://github.com/software-mansion/react-native-screens/assets/23281839/03aa7b3c-ffc7-430e-b3fe-bbf9eb7649f5">